sdk/servicebus/service-bus/src/core/receiverHelper.ts

```diff
@@ -85,11 +94,20 @@ export class ReceiverHelper {
   /**
    * Initiates a drain for the current receiver and resolves when
    * the drain has completed.
+   *
+   * NOTE: This method returns immediately if the receiver is not valid or if there
+   * are no pending credits on the receiver (ie: `receiver.credit === 0`).
    */
   async drain(): Promise<void> {
     const { receiver, logPrefix } = this._getCurrentReceiver();
 
     if (!this._isValidReceiver(receiver)) {
+      // TODO: should we throw?
```

Comment:
Yeah, this comment was me thinking "aloud". 

I think we could consider it a candidate to throw on but it seemed less important. The spot where it's used would basically do nothing useful if the receiver was invalid anyways but it'd be nice to go through here again. I think we have some future work to add in a "drain with timeout" for another bug in our tests, so perhaps we could lump in the investigation there.

Reply:
I would consider this a no-op and not an error scenario. The need for drain comes in order to avoid receiving messages when we are not ready for them. Absence of the receiver achieves that goal. So, I would combine the two checks here

```
if (!receiver || !received.isOpen() || receiver.credit === 0) {
   return;
   }
```